lib/Doctrine/DBAL/Driver/SQLSrv/SQLSrvConnection.php

```diff
@@ -140,7 +143,19 @@ public function exec($statement)
             throw SQLSrvException::fromSqlSrvErrors();
         }
 
-        return sqlsrv_rows_affected($stmt);
+        $affectedRowCount = sqlsrv_rows_affected($stmt);
+
+        $stmt = sqlsrv_query($this->conn, 'SELECT @@IDENTITY');
+
+        if ($statement !== false) {
+            sqlsrv_fetch($stmt);
+
+            $lastInsertId = sqlsrv_get_field($stmt, 0) ?: '0';
```

Comment:
I don't like`?:` here, it hides possible _error values_ (would that be `false`, `'0'` or `0` here?) - is it possible to use `!== false` for example?

Reply:
Same as above. Agree on improving error handling.